src/cli/remove.rs

```diff
@@ -44,22 +55,42 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     let results = deps
         .iter()
         .map(|dep| {
-            if let Some(p) = &args.platform {
-                project
-                    .manifest
-                    .remove_target_dependency(dep, &spec_type, p)
+            if args.pypi {
+                let pkg_name = rip::types::PackageName::from_str(dep.as_source())
+                    .expect("Expected dependency name.");
+                DependencyRemovalResult::PyPi(project.manifest.remove_pypi_dependency(&pkg_name))
             } else {
-                project.manifest.remove_dependency(dep, &spec_type)
+                DependencyRemovalResult::Conda(if let Some(p) = &args.platform {
+                    project
+                        .manifest
+                        .remove_target_dependency(dep, &spec_type, p)
+                } else {
+                    project.manifest.remove_dependency(dep, &spec_type)
+                })
             }
         })
-        .collect::<Vec<_>>();
+        .collect::<Vec<DependencyRemovalResult>>();
 
     project.save()?;
 
     // updating prefix after removing from toml
     let _ = get_up_to_date_prefix(&project, LockFileUsage::Update).await?;
 
-    for (removed, spec) in results.iter().flatten() {
+    for result in results.iter() {
+        let removed_and_spec = match result {
+            DependencyRemovalResult::Conda(Ok(pixi_result)) => {
+                (pixi_result.0.to_string(), pixi_result.1.to_string())
+            }
+            DependencyRemovalResult::PyPi(Ok(pypi_result)) => (
+                pypi_result.0.as_str().to_string(),
+                pypi_result.1.to_string(),
+            ),
+            DependencyRemovalResult::Conda(Err(e)) | DependencyRemovalResult::PyPi(Err(e)) => {
+                eprintln!("{e}");
+                return Ok(());
```

Comment:
The ``Ok`` was being returned before, so I continued that same behaviour.

Reply:
in what sense that is hard to read?

Reply:
I think the `Ok` is a previous mistake then. The problem with readability I see is that you could keep the `eprintln`s as one-liners in this match statement. So you don't need to go down to see what the output of this match is being used for.

Reply:
to be honest, I think this function should not return anything, the return seems to be a bit useless?
I believe it should let it ``panic`` in case of any problem

Reply:
Ok, gotcha, but the ``{}`` are needed since it execute two lines of code
to be honest I think that should ``return Err(e.into())``
or let it panic ``panic!("{e}")``

Reply:
I did some refactoring in this bit then